Lib/imaplib.py

```diff
@@ -284,23 +286,24 @@ def __exit__(self, *args):
     #       Overridable methods
 
 
-    def _create_socket(self):
+    def _create_socket(self, timeout):
         # Default value of IMAP4.host is '', but socket.getaddrinfo()
         # (which is used by socket.create_connection()) expects None
         # as a default value for host.
         host = None if not self.host else self.host
         sys.audit("imaplib.open", self, self.host, self.port)
-        return socket.create_connection((host, self.port))
+        return socket.create_connection((host, self.port), timeout)
 
-    def open(self, host = '', port = IMAP4_PORT):
+    def open(self, host = '', port = IMAP4_PORT,
+            timeout = socket._GLOBAL_DEFAULT_TIMEOUT):
```

Comment:
Again, default to `None` and adjust accordingly in the body. Although you might want to move that to `_create_socket` .